File: app/src/main/java/com/example/smartmemoryapp/AlarmReceiver.java
package com.example.smartmemoryapp;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class AlarmReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String taskName = intent.getStringExtra("task_name");
        
        Intent i = new Intent(context, OverlayActivity.class);
        i.putExtra("task_name", taskName);
        i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        context.startActivity(i);
    }
}

File: app/app/src/main/app/src/main/res/layout/app/src/main/java/com/smartmemory/app/app/src/main/java/com/smartmemory/app/AlarmReceiver.kt
package com.smartmemory.app

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class AlarmReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val note = intent.getStringExtra("NOTE") ?: "تنبيه!"
        
        val i = Intent(context, OverlayActivity::class.java)
        i.putExtra("NOTE", note)
        i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        i.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
        context.startActivity(i)
    }
}
